llama-index-core/llama_index/core/base/llms/types.py

```diff
@@ -118,6 +118,16 @@ def resolve_image(self, as_base64: bool = False) -> BytesIO:
             as_base64 (bool): whether the resolved image should be returned as base64-encoded bytes
 
         """
+        data_buffer = resolve_binary(
+            raw_bytes=self.image,
+            path=self.path,
+            url=str(self.url) if self.url else None,
+            as_base64=as_base64,
+        )
+        with data_buffer as r:
+            rb = r.read()
+        if len(rb) == 0:
+            raise ValueError("resolve_image returned zero bytes")
```

Comment:
We can do this a little better, and without calling `resolve_binary()` twice:

```suggestion
        data_buffer = resolve_binary(
            raw_bytes=self.image,
            path=self.path,
            url=str(self.url) if self.url else None,
            as_base64=as_base64,
        )
        
        # Check size by seeking to end and getting position
        data_buffer.seek(0, 2)  # Seek to end
        size = data_buffer.tell()
        data_buffer.seek(0)     # Reset to beginning
    
        if size == 0:
            raise ValueError("resolve_image returned zero bytes")
```

Reply:
Done with the last commit :))